openclsim/core.py

```diff
@@ -756,83 +772,96 @@ def __init__(self, criteria, *args, **kwargs):
         super().__init__(*args, **kwargs)
         """Initialization"""
         self.criteria = criteria
-        self.work_restrictions = {}
 
-    def calc_work_restrictions(self, location):
-        self.work_restrictions[location.name] = {}
+    def check_weather_restriction(self, location, event):
```

Comment:
Can we make this actually an abstract class?

Reply:
Change HasWeather to HasAbstractWeather and HasWorkabilityCriteria to HAsAbstractWorkabilityCriteria to be able to keep the original class names.
Further both classes do inherit now form ABC